### PR TITLE
Feat/upgrade terraform terragrunt providers modules

### PR DESCRIPTION
**Pull request by vitaliimd**
# PR Description
Updated terraform modules: acm, eventbridge, eks, route53, vpc, vpc-endpoints
Updated terraform version to 1.4.4 from 1.1.8 
Updated terragrunt version to 0.45.0 from 0.39.1
Updated terraform providers aws, kubernetes, helm, http, 
Fixes # (https://github.com/maddevsio/aws-eks-base/issues/326)

## Type of change

Delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### CODE REVIEW

File: terraform/layer1-aws/aws-eks.tf [hunk 1]
```diff
@@ -72,60 +63,10 @@ module "eks" {
 
   vpc_id = module.vpc.vpc_id
 
-  cluster_addons = local.eks_addons
-
-  cluster_encryption_config = var.eks_cluster_encryption_config_enable ? [
```
Comment (mglotov) [inline terraform/layer1-aws/aws-eks.tf:77]: Why has this configuration been deleted?

File: terraform/layer1-aws/aws-eks.tf [hunk 2]
```diff
@@ -72,60 +63,10 @@ module "eks" {
...
-      resources        = ["secrets"]
-    }
-  ] : []
-
   cluster_endpoint_public_access       = var.eks_cluster_endpoint_public_access
   cluster_endpoint_private_access      = var.eks_cluster_endpoint_private_access
   cluster_endpoint_public_access_cidrs = var.eks_cluster_endpoint_only_pritunl ? ["${module.pritunl[0].pritunl_endpoint}/32"] : ["0.0.0.0/0"]
 
-  # Extend cluster security group rules
-  cluster_security_group_additional_rules = {
```
Comment (mglotov) [inline terraform/layer1-aws/aws-eks.tf:89]: Why have these security groups been deleted?

File: terraform/layer1-aws/aws-eks.tf [hunk 3]
```diff
@@ -268,7 +207,7 @@ module "vpc_cni_irsa" {
 
 module "aws_ebs_csi_driver" {
   source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
-  version = "4.14.0"
+  version = "4.17.0"
```
Comment (mglotov) [inline terraform/layer1-aws/aws-eks.tf:210]: Why do you use different versions for the same module? See the `vpc_cni_irsa` terraform module

File: terraform/layer1-aws/aws-eks.tf [hunk 4]
```diff
@@ -282,32 +221,3 @@ module "aws_ebs_csi_driver" {
 
   tags = local.tags
 }
-
-resource "aws_kms_key" "eks" {
```
Comment (mglotov) [inline terraform/layer1-aws/aws-eks.tf:286]: Why has kms key been deleted?

File: terraform/layer1-aws/aws-eks.tf (hunk 2 above)
Comment (vitaliimd) [reply terraform/layer1-aws/aws-eks.tf:89]: in the eks module, they created groups and turned them on by default

File: terraform/layer1-aws/aws-eks.tf (hunk 4 above)
Comment (vitaliimd) [reply terraform/layer1-aws/aws-eks.tf:286]: in eks modules seted encryption for default

File: terraform/layer1-aws/aws-eks.tf (hunk 3 above)
Comment (vitaliimd) [reply terraform/layer1-aws/aws-eks.tf]: i'm sorry it's my mistake, fixed

File: terraform/layer1-aws/aws-eks.tf (hunk 1 above)
Comment (mglotov) [reply terraform/layer1-aws/aws-eks.tf:77]: For what kind of resources?

Comment (vitaliimd) [reply terraform/layer1-aws/aws-eks.tf:77]: for secrets

Comment (vitaliimd) [reply terraform/layer1-aws/aws-eks.tf:77]: https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/variables.tf#L105
https://github.com/terraform-aws-modules/terraform-aws-eks/blob/master/main.tf#L61

**Review by mglotov — APPROVED** [review]

**Review by khalezin — APPROVED** [review]

**Pull request merged**